parser/dsl/lexer.l

```diff
@@ -26,10 +28,31 @@
 
     // Code run each time yylex() is called.
     loc.step();
+
+    // A handy shortcut to the object used for capturing string literals.
+    auto& string_literal = drv.string_literal;
 %}
 
+'                      { BEGIN(ss_string_literal); string_literal.clear(); }
+<ss_string_literal>{
+    "\\'"              { string_literal.append(1, '\''); }
+    \\["]              { string_literal.append(1, '\"'); }
+    "\\?"              { string_literal.append(1, '\?'); }
+    "\\\\"             { string_literal.append(1, '\\'); }
+    "\\a"              { string_literal.append(1, '\a'); }
+    "\\b"              { string_literal.append(1, '\b'); }
+    "\\f"              { string_literal.append(1, '\f'); }
+    "\\n"              { string_literal.append(1, '\n'); }
+    "\\r"              { string_literal.append(1, '\r'); }
+    "\\t"              { string_literal.append(1, '\t'); }
+    "\\v"              { string_literal.append(1, '\v'); }
+    \\x[0-9]{3}        { string_literal.append(yytext); }
+    \\x[0-9A-Fa-f]+    { string_literal.append(yytext); }
+    '                  { BEGIN(INITIAL); return yy::parser::make_STRING_LITERAL(string_literal, loc); }
+    .                  { string_literal.append(1, *yytext); }
+    <<EOF>>            { return yy::parser::make_END_OF_INPUT(loc); }
```

Comment:
This should probably call `yyterminate` or `yyerror`.

Reply:
Throwing a Bison parser exception is a supported way to report syntax errors.

See https://www.gnu.org/software/bison/manual/bison.html#index-syntax_005ferror for details.

Reply:
nice find.